packages/synapse-react-client/src/components/QueryContext/QueryContext.tsx

```diff
@@ -23,11 +24,33 @@ export type LockedColumn = {
   value?: string
 }
 
-export type QueryContextType = {
+type OptionalQueryBundleRequestFields = keyof Pick<
+  QueryResultBundle,
+  | 'queryResult'
+  | 'queryCount'
+  | 'selectColumns'
+  | 'maxRowsPerPage'
+  | 'columnModels'
+  | 'facets'
+  | 'sumFileSizes'
+  | 'combinedSql'
+>
+
+export type QueryContextType<
+  TIncludedFields extends OptionalQueryBundleRequestFields = never,
+  ExcludeOtherFields extends true | false = false,
```

Comment:
These default generics maintain the existing behavior--all of the optional properties remain optional.